moped-editor/src/views/projects/ProjectTeamTable.js

```diff
@@ -28,193 +30,206 @@ const useStyles = makeStyles(theme => ({
   },
 }));
 
+const MEMBERS_QUERY = gql`
+  query Members {
+    moped_users(
+      order_by: { last_name: asc }
+      where: { status_id: { _eq: 1 } }
+    ) {
+      first_name
+      last_name
+      workgroup
+    }
+  }
+`;
+
+const ROLES_QUERY = gql`
+  query Roles {
+    moped_project_roles(order_by: { project_role_name: asc }) {
+      project_role_name
+    }
+  }
+`;
+
 const ProjectTeamTable = props => {
   const classes = useStyles();
 
+  const {
+    loading: membersLoading,
+    error: membersError,
+    data: members,
+  } = useQuery(MEMBERS_QUERY);
+
+  const { loading: roleLoading, error: roleError, data: roles } = useQuery(
+    ROLES_QUERY
+  );
+
   const handleAddRow = () => {
     let item = {
       id: props.StaffRows.length + 1,
       name: "",
       workgroup: "",
-      role: "",
+      role_name: "",
       notes: "",
     };
     props.setStaffRows(StaffRows => [...StaffRows, item]);
   };
 
   const handleRemoveRow = index => {
-    if (props.StaffRows.length > 1)
-      props.setStaffRows(props.StaffRows.slice(0, index));
-  };
-
-  const handleNameChange = (value, item, index) => {
-    item.name = value;
-    props.setStaffRows(props.StaffRows);
-  };
+    const updatedStaffRows = props.StaffRows.filter((row, i) => i !== index);
 
-  const handleRoleChange = (value, item, index) => {
-    item.role = value;
-    props.setStaffRows(props.StaffRows);
+    props.setStaffRows(updatedStaffRows);
   };
 
-  const handleGroupChange = (value, item, index) => {
-    item.workgroup = value.workgroup;
-    props.setStaffRows(props.StaffRows);
-  };
+  const handleNameAndWorkgroupChange = (value, index) => {
+    const updatedName = value?.name || null;
+    const updatedWorkgroup = value?.workgroup || null;
```

Comment:
ohhh! nice use of the optional chaining operator!! didn't realize we were already able do this with our current webpack/babel setup

Reply:
Same, I saw it somewhere in Devias so I just went with it. 👍